Dev: Document ellipsoid/MSL complexity and limitatoins (#6415)
ArduPilot/ardupilot_wiki: dev/source/docs/ros2-interfaces.rst

the GPS device reports data in Mean Sea Level (MSL). 
While the EGM-96 datum is common, certain GPS devices such as older 
UBLOX GPS's use an `approximated version that is not quite the same <https://portal.u-blox.com/s/question/0D52p00008l7VQACA2/what-is-the-egm96-grid-size>`_.

MAVLink also uses MSL for ``MAV_FRAME_GLOBAL``.
MAVROS converts MSL to WGS-84 ellipsoid height, but
`assumes EGM-96 datum <https://github.com/mavlink/mavros/blob/b49095727a6ff160e1e913b90a4ce50e383e8863/mavros/include/mavros/mavros_uas.hpp#L163>`_,
which is not the correct datum for all GPS's that ArduPilot supports.

According to the ``GeoPoint`` message documentation, height should always be above the WGS84 ellipsoid.
Due to resource constraints, ArduPilot can not automatically convert between the datums.

To enable this behavior to populate ``GeoPoint`` data with WGS-84 ellipsoid height, use the parameter :ref:`GPS_DRV_OPTIONS<GPS_DRV_OPTIONS>` to enable 
"Use ellipsoid height instead of AMSL". While this will allow ArduPilot to comply with the intent of
the portable WGS-84 ellipsoid model, it currently has known limitations:

> * Avoid using the default terrain data from ArduPilot's servers because it in EGM-96 geoid height if you want to fly low.

Reviewer: ```suggestion
* Avoid using the default terrain data from ArduPilot's servers, which is respect to EGM-96 geoid height, if you want to fly low.
```
